Chapter 1, Lesson 4 homework.

vytautas-sablinskas opened the pull request:
Checker in VS doesn't work, so I'll check it by using PR.

Commit:
First addition to homework 4.

Commit:
Completed homework, but VS checker doesn't work.

Commit:
Less errors while testing.

Commit:
Added \" \" to PromptInt.
Fixed mistake with if statement brackets in Bmi calculations.

Commit:
Updating and taking break, will need help on this.

Commit:
Fixed all possible errors 15/20, other 5 doesn't run so can't tell.

Commit:
Made floats return -1 if there's nothing written.

vytautas-sablinskas commented:
I have no clue honestly what to do right now, probably will watch Lesson 5 and then try to comeback because I'm stuck on this for long time already and Float checks don't show if they're correct or wrong.

Commit:
Figured out my stupid mistake which made it not run. 20/20 passed so it should be okay.

vytautas-sablinskas commented:
Done all checks, I think it should be alright now.

Almantask commented on Src/BootCamp.Chapter/Lesson4.cs:
```diff
@@ -0,0 +1,106 @@
+        public static void Demo()
+        {
+            for (int i = 0; i < 2; i++)
+            {
```
The body of for loop could be moved into a function. That would leave you with two options then:
A- just call the same function twice
B- do what you do, but in a body call a function instead.

Almantask commented on Src/BootCamp.Chapter/Lesson4.cs:
```diff
@@ -0,0 +1,106 @@
+            bool isValidWeight = true;
+            bool isValidHeight = true;
+
+            if (height <= 0)
```
Can be simplified:
```cs
string error = "";
if(weight <= 0)
{
 error = ...;
}
else if(height <= 0)
{
  error += "";
}
if(string.IsNullOrEmpty(error))
{
  print(error);
  return -1;
}
else
{
  return bmi;
}
```

Almantask reviewed (changes requested):
A bit overly complex validation

Commit:
In demo function I call another function AskingInformation.
I've tried simplifying it, but it's really hard and I'm not sure if I've done it right.

vytautas-sablinskas commented:
Okay, so I redid it a bit. Now I added another function AskingInformation and called it in Demo function. 
Then I tried simplifying this thing, but not sure if I got it right. It's pretty hard to understand this, but I hope I can get better at it.

Almantask reviewed (approved):
Looks better 😄
